tests/test_aside.py

```diff
@@ -67,3 +78,69 @@ def test_toggle_block_open_staff_only(self, is_staff, expected_status):
         with patch.object(self.aside_instance, 'is_staff', return_value=is_staff):
             resp = self.aside_instance.toggle_block_open_status()
         assert resp.status_code == expected_status
+
+    @data(*[
+        [True, 200],
+        [False, 403]
+    ])
+    @unpack
+    def test_responses_staff_only(self, is_staff, expected_status):
+        """
+        Test that only staff users should access the API
+        """
+        with patch.object(self.aside_instance, 'is_staff', return_value=is_staff):
+            resp = self.aside_instance.responses()
+        assert resp.status_code == expected_status
+
+    @data(True, False)
+    def test_responses_open(self, is_open):
+        """
+        Test that the responses API shows whether the problem is open
+        """
+        RapidResponseBlockStatus.objects.create(
+            usage_key=self.aside_instance.wrapped_block_usage_key,
+            course_key=self.aside_instance.course_key,
+            open=is_open,
+        )
+
+        resp = self.aside_instance.responses()
+        assert resp.status_code == 200
+        assert resp.json['is_open'] == is_open
+
+    @pytest.mark.usefixtures("example_event")
+    def test_responses(self):
+        """
+        Test that the responses API will show recorded events during the open period
+        """
+        event = self.example_event
+        event_before = dict_with(event, {'test_data': 'before'})
+        event_during = dict_with(event, {'test_data': 'during'})
+        event_after = dict_with(event, {'test_data': 'after'})
+
+        problem_id = UsageKey.from_string(event['event']['problem_id'])
+        course_id = CourseLocator.from_string(event['context']['course_id'])
+
+        recorder = SubmissionRecorder()
+        recorder.send(event_before)
+        block_status = RapidResponseBlockStatus.objects.create(
+            usage_key=problem_id,
+            course_key=course_id,
+            open=True,
+        )
+        recorder.send(event_during)
+        block_status.open = False
+        block_status.save()
+        recorder.send(event_after)
+
+        assert RapidResponseSubmission.objects.count() == 1
+        submission = RapidResponseSubmission.objects.first()
+        assert submission.event['test_data'] == event_during['test_data']
+
+        resp = self.aside_instance.responses()
+        assert resp.status_code == 200
+        assert resp.json['is_open'] is False
+        assert resp.json['responses'] == [{
+            'id': submission.id,
+            'answer_id': submission.answer_id,
+            'answer_text': submission.answer_text,
+        }]
```

Comment:
We should have separate test cases for (1) ensuring that responses are only recorded during the 'open' period, and (2) the `responses` handler method

Reply:
Makes sense